### PR TITLE
Update statistical insights guide and fix warning

### LINKED ISSUE #606: Documentation - Update Gain Statistical Insights into Your Data Guide
Update document to use new accessor approach: https://woodwork.alteryx.com/en/stable/guides/statistical_insights.html

### LINKED ISSUE #665: Accessor: Running `mutual_information` generates a number of `SettingWithCopyWarning` warnings
When running `df.ww.mutual_information()` a number of `SettingWithCopyWarning` warnings are generated by pandas. This behavior also appears to be present on the DataTable implementation and is present in our docs. It would be nice to fix this for the accessor implementation.

See existing docs for example error:
https://woodwork.alteryx.com/en/stable/guides/statistical_insights.html#DataTable.mutual_information

### PR DESCRIPTION
**Pull request by thehomebrewnerd**
- Update statistical insights guide and fix warning
- Closes #606
- Closes #665 

Updates statistical insights guide to use new accessor approach. Also updates mutual information code to prevent pandas warning from appearing in documentation.

### CODE REVIEW

File: docs/source/guides/statistical_insights.ipynb [hunk 1]
```diff
@@ -4,11 +4,11 @@
...
    "metadata": {},
    "source": [
-    "# Gain Statistical Insights into Your DataTable\n",
+    "# Gain Statistical Insights into Your Data\n",
     "\n",
-    "Woodwork provides methods on DataTable to allow you to use the typing information inherent in a DataTable to better understand your data.\n",
+    "Woodwork provides methods on your dataframes to allow you to use the typing information stored by Woodwork to better understand your data.\n",
     "\n",
-    "Follow along to learn how to use `describe` and `mutual_information` on a retail DataTable so that you can see the full capabilities of the functions."
+    "Follow along to learn how to use `describe` and `mutual_information` on a Woodwork initialized dataframe of retail data so that you can see the full capabilities of the functions."
```
Comment (tamargrey) [inline docs/source/guides/statistical_insights.ipynb:11]: Should probably also include `value_counts` in this sentence

File: docs/source/guides/statistical_insights.ipynb [hunk 2]
```diff
@@ -90,29 +89,29 @@
...
     "    - Defaults to using 10 bins\n",
     "    - The more bins there are, the more variety a column will have. The number of bins used should accurately portray the spread of the data.\n",
-    "- `nrows` - If `nrows` is set at a value below the number of rows in the DataTable, that number of rows is randomly sampled from the underlying data\n",
+    "- `nrows` - If `nrows` is set at a value below the number of rows in the DataFrame, that number of rows is randomly sampled from the underlying data\n",
     "    - Defaults to using all the available rows.\n",
-    "    - Decreasing the number of rows can speed up the mutual information calculation on a DataTable with many rows, but you should be careful that the number being sampled is large enough to accurately portray the data."
+    "    - Decreasing the number of rows can speed up the mutual information calculation on a DataFrame with many rows, but you should be careful that the number being sampled is large enough to accurately portray the data."
```
Comment (tamargrey) [inline docs/source/guides/statistical_insights.ipynb:107]: We'll need to remember to add the `include_index` parameter here when we add it into the Accessor in #694

**Review by tamargrey — COMMENTED** [review]
Looking good! Might be better to wait to get this in until #694 (and maybe #688 but that probably doesn't need to be referenced in the guide) get in so we can display that behavior

File: docs/source/guides/statistical_insights.ipynb (hunk 1 above)
Comment (thehomebrewnerd) [reply docs/source/guides/statistical_insights.ipynb:11]: How about if we make this more generic instead, so we won't have to remember to update all the time if we add more methods?

File: docs/source/guides/statistical_insights.ipynb (hunk 2 above)
Comment (thehomebrewnerd) [reply docs/source/guides/statistical_insights.ipynb:107]: Was going to update the issue, but see you already have it covered!

File: docs/source/guides/statistical_insights.ipynb (hunk 1 above)
Comment (tamargrey) [reply docs/source/guides/statistical_insights.ipynb:11]: that sounds good to me!

Comment (thehomebrewnerd) [reply docs/source/guides/statistical_insights.ipynb]: Cleaned up the wording of this sentence and removed mentions of specific methods.

**Review by jeff-hernandez — APPROVED** [review]
Looks good!

**Pull request merged**